daml-script/test/src/test-utils/com/daml/lf/engine/script/test/MultiParticipantFixture.scala

```diff
@@ -99,13 +104,10 @@ trait MultiParticipantFixture
                 )
               ).run()
             )
-            _ <- Future.traverse(
-              Seq(p1LedgerApi, p2LedgerApi, p1AdminApi, p2AdminApi, domainPublicApi, domainAdminApi)
-            )(p =>
-              RetryStrategy.constant(attempts = 120, waitTime = 1.seconds)((_, _) =>
-                Future(p.testAndUnlock(InetAddress.getLoopbackAddress))
-              )
-            )
+
+            _ <- RetryStrategy.constant(attempts = 240, waitTime = 1.seconds) { (_, _) =>
+              Future(Files.size(portFile))
```

Comment:
Just for my own clarity, why are we sure that the addresses will exist before the port file? Or if we're not, why don't we need to wait for that anymore?

Reply:
canton garantees the ports are fully functional (not only listening) before it write the portfile. 
It is the official way to be sure it is up and running.